src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addproperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
+        action.keysToRemove.forEach(key => {
+          delete currentState[key];
+        });
+        break;
+      default:
+        throw new Error('Unknown action Type: ${action.type}');
```

Comment:
The error message uses template literals incorrectly. It should be `throw new Error(`Unknown action Type: ${action.type}`);` with backticks instead of single quotes.